src/test/java/org/jboss/windup/rules/tests/WindupRulesLinksTest.java

```diff
@@ -152,8 +154,20 @@ private boolean isValidLink(final String link)
                 final HttpURLConnection urlConn = (HttpURLConnection) url.openConnection();
                 // property name from https://docs.oracle.com/javase/8/docs/technotes/guides/net/properties.html
                 urlConn.setConnectTimeout(Integer.getInteger("sun.net.client.defaultConnectTimeout", 5000));
-                urlConn.connect();
-                CACHE_ANALYZED_LINKS.put(link, urlConn.getResponseCode());
+                int attempt = 0;
+                while (true)
```

Comment:
I would recommend not to use this infinite loops. Even you are not specifying the delay between calls, so this is an over-kill.
You can use [Awaitily](https://github.com/awaitility/awaitility) to do this kind of checks where you have a max time or max retries to consider a process Success, and also you can define the delay between attempts. 
We already use that in xavier-integration/EndToEndTests : https://github.com/project-xavier/xavier-integration/blob/9b01b4c1a932c16976b38f476a41a0d2bb799bcd/src/test/java/org/jboss/xavier/integrations/EndToEndTest.java#L596